summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -175,6 +185,10 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
         case OptionMappedProducer(_, op) =>
           val boxed = Externalizer(op)
           spout.flatMap { case (time, t) => boxed.get(t).map { x => (time, x) } }
+        case FlatMappedProducer(_, op) => {
+          val lockedOp = Externalizer(op)
+          spout.flatMap { case (time, t) => lockedOp.get.apply(t).map { x => (time, x) } }
```

Comment:
The OptionMappedProducer and FlatMappedProducer cases look exactly the same. may be combine them like this:
case OptionMappedProducer(_, op)  | FlatMappedProducer(_, op) =>